src/Runtime/Eval.hs

```diff
@@ -184,15 +185,15 @@ eval (App n es) = do
     Nothing -> do
       f <- lookupName n
       case f of
-        Just (Vf params env' e) -> extScope (zip params (args) ++ env') (evalWithLimit (eval e)) -- ++ env?
-        Just (Pv env' e)        -> extScope (zip [] (args) ++ env') (evalWithLimit (eval e)) -- ++ env?
+        Just (Vf params env' e) -> extScope (extendEvalEnv (params,args) env') (evalWithLimit (eval e))--(Map.union (Map.fromList (zip params (args))) env') (evalWithLimit (eval e)) -- ++ env?
+        Just (Pv env' e)        -> extScope (extendEvalEnv ([],args) env') (evalWithLimit (eval e)) --(Map.union (Map.fromList (zip [] (args))) env') (evalWithLimit (eval e)) -- ++ env?
```

Comment:
The transformation looks appropriate, but I am a bit confused on why we were zipping an empty list with arguments in the first place. Or why a pending value appears in the App case in the first place. It shouldn't change in this commit, but I will look over it again to figure out what's happening after it is merged.